src/main/cpp/include/genomicsdb/genomicsdb_columnar_field.h

```diff
@@ -0,0 +1,326 @@
+    inline void set_num_live_entries(const size_t n)
+    {
+      m_num_live_entries = n;
+      m_num_filled_entries = n;
+      m_num_unprocessed_entries = n;
+    }
+    inline void decrement_num_live_entries(const size_t value)
+    {
+      assert(m_num_live_entries >= value);
+      m_num_live_entries -= value;
+    }
+    inline void increment_num_live_entries(const size_t val=1u)
+    {
+      m_num_live_entries += val;
+    }
+    inline size_t get_num_unprocessed_entries() const
+    {
+      return m_num_unprocessed_entries;
+    }
+    inline void decrement_num_unprocessed_entries(const size_t value)
+    {
+      assert(m_num_unprocessed_entries >= value);
+      m_num_unprocessed_entries -= value;
+    }
+    inline std::vector<uint8_t>& get_buffer() { return m_buffer; }
+    inline const std::vector<uint8_t>& get_buffer() const { return m_buffer; }
+    inline uint8_t* get_buffer_pointer() { return &(m_buffer[0]); }
+    inline const uint8_t* get_buffer_pointer() const { return &(m_buffer[0]); }
+    inline size_t get_buffer_size_in_bytes() const { return m_buffer.size(); }
+    /*inline std::vector<size_t>& get_offsets() { return m_offsets; }*/
+    inline size_t* get_offsets_pointer() { return &(m_offsets[0]); }
+    inline size_t get_offsets_size_in_bytes() const { return ((m_offsets.size()-1u)*sizeof(size_t)); } //last offsets element holds the "size"
+    inline size_t get_offsets_length() const { return (m_offsets.size()-1u); } //last offsets element holds the "size"
+    /*
+     * Set offset in buffer of the idx-th element
+     * Useful only for variable length fields and used by the iterators
+     * This allows us to use m_offsets for computing the size of every cell data without
+     * having to write "if(last_cell) else" expressions
+     */
+    inline void set_offset(const size_t idx, const size_t value)
+    {
+      assert(idx < m_offsets.size());
+      m_offsets[idx] = value;
+    }
+    inline size_t get_offset(const size_t idx) const
+    {
+      assert(idx < m_offsets.size() && idx < m_num_filled_entries);
+      return m_offsets[idx ];
+    }
+    inline size_t get_size_of_variable_length_field(const size_t idx) const
+    {
+      assert(idx+1u < m_offsets.size() && idx < m_num_filled_entries);
+      return m_offsets[idx+1u] - m_offsets[idx];
+    }
+    inline bool is_valid(const size_t idx) const
+    {
+      assert(idx < m_valid.size() && idx < m_num_filled_entries);
+      return m_valid[idx];
+    }
+    inline std::vector<bool>& get_valid_vector() { return m_valid; }
+    //Buffer pointers
+    inline void set_next_buffer(GenomicsDBBuffer* next) { m_next_buffer = next; }
+    inline void set_previous_buffer(GenomicsDBBuffer* previous) { m_previous_buffer = previous; }
+    inline GenomicsDBBuffer* get_next_buffer() { return m_next_buffer; }
+    inline GenomicsDBBuffer* get_previous_buffer() { return m_previous_buffer; } 
+  protected:
```

Comment:
ignore this. I was confused because msg_ in exception class had a trailing underscore